### ISSUE TITLE
Nix formatting of multiline codeblocks is broken

### ISSUE THREAD

**Issue by sumnerevans** [vector-im/element-web#18871]
### Steps to reproduce

Send a message with the following content to a room:

    ```nix
    {
      foo = ''
        bar
      '';
    }
    ```

### What happened?

![image](https://user-images.githubusercontent.com/16734772/131870634-0f3ba26b-0ca9-4ec3-8f45-2994e1ac075b.png)

### What did you expect?

I expect that the code block is shown correctly.

### Operating system

NixOS

### Application version

Element version: 1.8.2, Olm version: 3.2.3

### How did you install the app?

https://github.com/NixOS/nixpkgs/pull/136311

### Homeserver

sumnerevans.com

### Have you submitted a rageshake?

No

**Comment by sumnerevans** [comment]
I do have "Enable automatic language detection for syntax highlighting" on, but it also persists when it's disabled.

Here is the decrypted source for the message:

```
{
  "room_id": "!MahzEgrijBRaMQLBRS:matrix.org",
  "type": "m.room.message",
  "content": {
    "m.new_content": {
      "msgtype": "m.text",
      "body": "```nix\n{\n  foo = ''\n    bar\n  '';\n}\n```",
      "format": "org.matrix.custom.html",
      "formatted_body": "<pre><code class=\"language-nix\">{\n  foo = ''\n    bar\n  '';\n}\n</code></pre>\n"
    },
    "msgtype": "m.text",
    "body": " * ```nix\n{\n  foo = ''\n    bar\n  '';\n}\n```",
    "format": "org.matrix.custom.html",
    "formatted_body": " * <pre><code class=\"language-nix\">{\n  foo = ''\n    bar\n  '';\n}\n</code></pre>\n"
  }
}
```

**Comment by robintown** [comment]
Fixed by https://github.com/matrix-org/matrix-react-sdk/pull/6681 (I mistakenly thought that this fix was already in 1.8.2, hence my confusion)
